src/app/api/oneproduct.service.ts

```diff
@@ -0,0 +1,50 @@
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable, catchError, throwError } from 'rxjs';
+import { IOneSingleProduct } from '../Interface/singleproduct';
+import { CountryService } from '../services/country.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class OneProductService {
+  oneProduct!: IOneSingleProduct | undefined;
+
+  private oneProductUrl =
+    'https://ac-trainee-store-api.herokuapp.com/products/';
+
+  private oneProductSubject: BehaviorSubject<IOneSingleProduct | undefined> =
```

Comment:
Would it be okay for you to rename this to productSubject?

Reply:
oneProductSubject was renamed to productSubject